zig/shell.nix

```diff
@@ -0,0 +1,67 @@
+{ pkgs ? import <nixpkgs> {} } :
+let
+	# Right. Getting this to work took more effort than all the other shells put together.
+	# Things that didn't work:
+	# - Directly using the Zig package, pinned to any version. Nix is "known as the most up to date distribution", but this project uses a *nightly* build of Zig, so that's not there.
```

Comment:
I think now that 0.10 has been released, we might be ok with that? Back when 0.9 was stable I needed to use nightly to work around some compiler issues...

Reply:
Unfortunately, I kept on getting some kind of type error to do with `clap` when using the official 0.10 release. I still have my terminal open from that. I think this is it:

```
$ ./run.sh --headless --silent --profile 3 --turbo ../../opus5.gb 
/home/user/RosettaBoy/nix/zig/build.zig:23:32: error: expected type 'builtin.ExportOptions', found 'type'
    exe.addPackagePath("clap", "lib/clap/clap.zig");
                               ^~~~~~~~~~~~~~~~~~~
/nix/store/dypjlm6wk41gb0fdb8bbdshs5qvbhmx5-zig-0.10.0/lib/std/builtin.zig:662:27: note: struct declared here
pub const ExportOptions = struct {
                          ^~~~~~
referenced by:
    runBuild: /nix/store/dypjlm6wk41gb0fdb8bbdshs5qvbhmx5-zig-0.10.0/lib/build_runner.zig:229:45
    usage__anon_5122: /nix/store/dypjlm6wk41gb0fdb8bbdshs5qvbhmx5-zig-0.10.0/lib/build_runner.zig:240:13
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

IIRC I tried some stuff to fix this (hence wondering whether I was syncing the submodules correctly), but as I don't know any Zig, eventually I gave up and figured going with the nightly would be easier…